packages/perseus/src/util/graphie.test.ts

```diff
@@ -0,0 +1,1251 @@
+        it("uses the style, if given", () => {
+            const graphie = createAndInitGraphie();
+            const mockRaphaelElement = createMockRaphaelElement();
+            graphie.raphael.path.mockReturnValue(mockRaphaelElement);
+
+            graphie.line([0, 0], [1, 1], {stroke: "#112233"});
+
+            expect(mockRaphaelElement.attr).toHaveBeenCalledWith(
+                expect.objectContaining({
+                    stroke: "#112233",
+                }),
+            );
+        });
+
+        it("uses the default style, if none given", () => {
+            const graphie = createAndInitGraphie();
+            const mockRaphaelElement = createMockRaphaelElement();
+            graphie.raphael.path.mockReturnValue(mockRaphaelElement);
+
+            graphie.line([0, 0], [1, 1]);
+
+            expect(mockRaphaelElement.attr).toHaveBeenCalledWith({
+                fill: "none",
+                "stroke-width": 2,
+            });
+        });
+
+        it("restores the previous style after drawing", () => {
+            const graphie = createAndInitGraphie();
+            const el1 = createMockRaphaelElement("el1");
+            const el2 = createMockRaphaelElement("el2");
+            graphie.raphael.path.mockReturnValueOnce(el1);
+            graphie.raphael.path.mockReturnValueOnce(el2);
+
+            graphie.line([0, 0], [1, 1], {stroke: "#112233"});
+            graphie.line([0, 0], [1, 1]);
+
+            expect(el2.attr).toHaveBeenCalledWith({
+                fill: "none",
+                "stroke-width": 2,
+            });
+        });
+
+        it("dasherizes Raphael attribute names (e.g. strokeWidth -> stroke-width)", () => {
+            const graphie = createAndInitGraphie();
+            const mockRaphaelElement = createMockRaphaelElement();
+            graphie.raphael.path.mockReturnValue(mockRaphaelElement);
+
+            graphie.line([0, 0], [1, 1], {strokeWidth: 42});
+
+            expect(mockRaphaelElement.attr).toHaveBeenCalledWith(
+                expect.objectContaining({
+                    "stroke-width": 42,
+                }),
+            );
+        });
+
+        it("returns the Raphael element", () => {
+            const graphie = createAndInitGraphie();
+            const mockRaphaelElement = createMockRaphaelElement();
+            graphie.raphael.path.mockReturnValue(mockRaphaelElement);
+
+            const result = graphie.line([0, 0], [1, 1]);
+
+            expect(result).toBe(mockRaphaelElement);
+        });
+    });
+
+    describe("parabola", () => {
+        it("draws a path using the quadratic 'Q' SVG command", () => {
+            const graphie = createAndInitGraphie();
+            const mockRaphaelElement = createMockRaphaelElement();
+            graphie.raphael.path.mockReturnValue(mockRaphaelElement);
+
+            graphie.parabola(1, 2, 3);
+
+            expect(graphie.raphael.path).toHaveBeenCalledWith(
+                expect.stringContaining("Q"),
+            );
```

Comment:
non-blocking: I wonder if it is it possible to assert a start or end point at least?  🤔